packages/opencensus-core/src/resource/resource.ts

```diff
@@ -0,0 +1,192 @@
+
+/**
+ * Resource represents a resource, which capture identifying information about
+ * the entities for which signals (stats or traces) are reported. It further
+ * provides a framework for detection of resource information from the
+ * environment and progressive population as signals propagate from the core
+ * instrumentation library to a backend's exporter.
+ */
+export class Resource {
+  /**
+   * The type identifier for the resource.
+   */
+  private readonly type: string|null;
+  /**
+   * A map of labels that describe the resource.
+   */
+  private readonly labels: StringMap;
+
+  private static readonly MAX_LENGTH = 255;
+  private static readonly LABEL_LIST_SPLITTER = ',';
+  private static readonly LABEL_KEY_VALUE_SPLITTER = '=';
+  private static readonly ENV_TYPE =
+      Resource.parseResourceType(process.env.OC_RESOURCE_TYPE);
+  private static readonly ENV_LABEL_MAP =
+      Resource.parseResourceLabels(process.env.OC_RESOURCE_LABELS);
+  private static readonly ERROR_MESSAGE_INVALID_CHARS =
+      'should be a ASCII string with a length greater than 0 and not exceed ' +
+      Resource.MAX_LENGTH + ' characters.';
+  private static readonly ERROR_MESSAGE_INVALID_VALUE =
+      'should be a ASCII string with a length not exceed ' +
+      Resource.MAX_LENGTH + ' characters.';
+
+  constructor(type: string, labels: StringMap) {
+    this.type = type;
+    this.labels = labels;
+  }
+
+  /**
+   * Returns a Resource. This resource information is loaded from the
+   * OC_RESOURCE_TYPE and OC_RESOURCE_LABELS environment variables.
+   *
+   * @returns {Resource}
+   */
+  static createFromEnvironmentVariables(): Resource {
+    return new Resource(Resource.ENV_TYPE, Resource.ENV_LABEL_MAP);
+  }
+
+  /**
+   * Returns a Resource that runs all input resources sequentially and merges
+   * their results. In case a type of label key is already set, the first set
+   * value takes precedence.
+   *
+   * @param  {Resource[]} resources the list of the resources.
+   * @returns {Resource}
+   */
+  static mergeResources(resources: Resource[]): Resource {
+    let currentResource: Resource = null;
+    for (const resource of resources) {
+      currentResource = this.merge(currentResource, resource);
+    }
+    return currentResource;
+  }
+
+  /**
+   * Returns the type identifier for the resource.
+   *
+   * @returns {string}
+   */
+  getType(): string {
+    return this.type;
+  }
+
+  /**
+   * Returns a map of labels that describe the resource.
+   *
+   * @returns {StringMap}
+   */
+  getLabels(): StringMap {
+    return this.labels;
+  }
+
+  /**
+   * Creates a resource type from the OC_RESOURCE_TYPE environment variable.
+   *
+   * <p>OC_RESOURCE_TYPE: A string that describes the type of the resource
+   * prefixed by a domain namespace, e.g. “kubernetes.io/container”.
+   */
+  private static parseResourceType(rawEnvType: string): string {
+    if (rawEnvType && rawEnvType != null) {
```

Comment:
Maybe this line can be replaced with simply
```javascript
if (rawEnvType)  // works in Javascript
```
though I'm not very familiar with Typescript. /cc @isaikevych

Reply:
Done.